src/Entity/Membership.php

```diff
@@ -248,6 +252,15 @@ public function postCreate(EntityStorageInterface $storage) {
     parent::postCreate($storage);
   }
 
+  /**
+   * @inheritDoc
+   */
+  function postSave(EntityStorageInterface $storage, $update = TRUE) {
+    $event = new MembershipEvent($this);
+    $this->getEventDispatcher()->dispatch(MembershipEvents::UPDATED, $event);
+    parent::postSave($storage, $update);
+
+  }
```

Comment:
Ok. Still haven't gotten to the state change events yet -- probably will be in there today or tomorrow -- so I haven't seen how state_machine handles this.

I was trying to avoid hook_entity_insert() and hook_entity_update() and go with a more OO approach, and this looked like the best way...

What's your sense of preference going forward -- symfony events vs hooks? I would prefer events, just because that's a pattern I use on a bunch of projects (and never particularly liked hooks). But I can live with a hook here if they aren't going to be deprecated... I was surprised to not find any decorator or observers related to the Entity Manager to supersede these hooks...